stdlib/public/core/Integers.swift.gyb

```diff
@@ -3247,208 +3294,348 @@ public struct DoubleWidth<T : FixedWidthInteger>
+      lhs = DoubleWidth((High(0), Low(extendingOrTruncating: lhs._storage.high)))
+      return
+    }
+
+    lhs._storage.low >>= rhs._storage.low
+    if Base.bitWidth > rhs._storage.low {
+      lhs._storage.low |= Low(extendingOrTruncating: lhs._storage.high <<
+        numericCast(numericCast(Base.bitWidth) - rhs._storage.low))
+    } else {
+      lhs._storage.low |= Low(extendingOrTruncating: lhs._storage.high >>
+        numericCast(rhs._storage.low - numericCast(Base.bitWidth)))
+    }
+    lhs._storage.high >>= High(extendingOrTruncating: rhs._storage.low)
+  }
+  
 % for x in chain(*binaryArithmetic.values()):
 
   // FIXME(integers): remove this once the operators are back to Numeric
   public static func ${x.operator} (
-    lhs: DoubleWidth<T>, rhs: DoubleWidth<T>
-  ) -> DoubleWidth<T> {
+    lhs: DoubleWidth, rhs: DoubleWidth
+  ) -> DoubleWidth {
     var lhs = lhs
     lhs ${x.operator}= rhs
     return lhs
   }
 
+%   argumentLabel = (x.firstArg + ':') if x.firstArg != '_' else ''
   public static func ${x.operator}=(
-    lhs: inout DoubleWidth<T>, rhs: DoubleWidth<T>
+    lhs: inout DoubleWidth, rhs: DoubleWidth
   ) {
-    fatalError()
+    let (result, overflow) = lhs.${x.name}ReportingOverflow(${argumentLabel}rhs)
+    _precondition(overflow == .none, "Overflow in ${x.operator}=")
+    lhs = result
   }
 % end
 
   public init(_truncatingBits bits: UInt) {
-    fatalError()
+    _storage.low = Low(_truncatingBits: bits)
+    _storage.high = High(_truncatingBits: bits >> UInt(Base.bitWidth))
   }
 
   // other
   //
   public init(_builtinIntegerLiteral x: _MaxBuiltinIntegerType) {
-    fatalError("Method must be overridden")
+    // FIXME: This obviously won't work if Int(x) fails
+    self.init(Int(_builtinIntegerLiteral: x))
   }
 
   public var description: String {
     return "(\(_storage.high), \(_storage.low))"
   }
 
   public var leadingZeroBitCount: Int {
-    fatalError()
+    return high == 0 as High
+      ? Base.bitWidth + low.leadingZeroBitCount
+      : high.leadingZeroBitCount
   }
 
   public var trailingZeroBitCount: Int {
-    fatalError()
+    return low == 0 as Low
+      ? Base.bitWidth + high.trailingZeroBitCount
+      : low.trailingZeroBitCount
   }
 
   public var nonzeroBitCount: Int {
-    fatalError()
+    return high.nonzeroBitCount + low.nonzeroBitCount
   }
 
   public var hashValue: Int {
-    fatalError()
+    return _mixInt(high.hashValue) ^ low.hashValue
   }
 
-  public init(littleEndian: DoubleWidth<T>) {
-    fatalError()
+  public init(littleEndian value: DoubleWidth) {
+#if _endian(little)
+    self = value
+#else
+    self = value.byteSwapped
+#endif
   }
 
-  public init(bigEndian: DoubleWidth<T>) {
-    fatalError()
+  public init(bigEndian value: DoubleWidth) {
+#if _endian(big)
+    self = value
+#else
+    self = value.byteSwapped
+#endif
   }
 
-  public var littleEndian: DoubleWidth<T> {
-    fatalError()
+  public var littleEndian: DoubleWidth {
```

Comment:
It is! Factored that out into a `FixedWidthInteger` default implementation—`byteSwapped` is the customization point. I didn't remove the implementations from the standard integer types.